lib/pluginmanager/install.rb

```diff
@@ -43,6 +43,7 @@ def validate_cli_options!
   # Check if the specified gems contains
   # the logstash `metadata`
   def verify_remote!(gems)
+    Gem.sources = Gem::SourceList.from(gemfile.gemset.sources)
     gems.each do |plugin, version|
       puts("Validating #{[plugin, version].compact.join("-")}")
       signal_error("Installation aborted, verification failed for #{plugin} #{version}") unless LogStash::PluginManager.logstash_plugin?(plugin, version)
```

Comment:
does signal_error terminate the block?
otherwise it would be possible to validate all then signal all errors e.g.
```ruby
failed = []
gems.each do |plugin, version|
  next if ...
  failed << "verification failed for #{plugin} #{version}"
end
msg = failed.join(', ')
signal_error msg.prepend('Installation aborted,  ')
```